qualang_tools/video_mode/videomode.py

```diff
@@ -0,0 +1,452 @@
+# VideoMode script enabling the update of parameters in a QUA program through user input while the program is running.
+# Authors: Arthur Strauss, Théo Laudat
+# Date: 19/12/2023
+
+import threading
+from qm.qua import *
+from qm import QuantumMachine, QmJob, Program
+import time
+import signal
+from typing import Optional, List, Dict, Union
+from dataclasses import dataclass
+import numpy as np
+
+
+@dataclass
+class ParameterTable:
+    """
+    Data class enabling the mapping of parameters to be updated to their corresponding "to-be-declared" QUA variables.
+    The type of the QUA variable to be adjusted is automatically inferred from the type of the initial_parameter_value.
+    Each parameter in the dictionary should be given a name that the user can then easily access through the table with
+    table[parameter_name]. Calling this will return the QUA variable built within the QUA program corresponding to the parameter name
+    and its associated Python initial value.
+    Args:
+        parameters_dict: Dictionary of the form { "parameter_name": initial_parameter_value }.
+        the QUA program.
+    """
+
+    parameters_dict: Dict[str, Union[float, int, bool, List, np.ndarray]]
+
+    def __post_init__(self):
+        self.table = {}
+        for index, (parameter_name, parameter_value) in enumerate(
+            self.parameters_dict.items()
+        ):
+            self.table[parameter_name] = {"index": index, "value": parameter_value}
+            if isinstance(parameter_value, float):
+                if float(parameter_value).is_integer() and parameter_value > 8:
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(int, value=int({parameter_value}))"
+                    self.table[parameter_name]["type"] = int
+                else:
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(fixed, value={parameter_value})"
+                    self.table[parameter_name]["type"] = float
```

Comment:
Use a class for the declaration data, don't use dictionaries

Reply:
Do you mean for `param_dict`?

Reply:
Confusing for me too, you would like to define a class for each single parameter with the attributes corresponding to var, type, index ?